docs/advanced/test-command.md

```diff
@@ -0,0 +1,91 @@
+
+To be able to establish direct connection, you have to ensure:
+
+- Your machine is publicly accessible on the internet or atleast a specific port is.
+- You add flag `p2p-tcp-address` (i.e.: `127.0.0.1:9001`) flag and the port specified in it is free and publicly accessible.
+- You add the flag `p2p-external-ip` (i.e.: `8.8.8.8`) and specify your public ip.
+
+If all points are satisfied by you and the other peers, you should be able to establish a direct TCP connection between each other. Note that a relay is still required, as establishing direct connection is facilitated by it.
+
+### Pre-requisites
+
+- [Create ENR](http://localhost:3000/docs/next/charon/charon-cli-reference#creating-an-enr-for-charon).
+- Share your ENR with other peers which will test against you.
+- Obtain the ENRs of the other peers against which you will test.
+
+### Run
+
+```shell
+charon alpha test peers \
+  --enrs="enr:-HW4QNDXi9MzdH9Af65g20jDfelAJ0kJhclitkYYgFziYHXhRFF6JyB_CnVnimB7VxKBGBSkHbmy-Tu8BJq8JQkfptiAgmlkgnY0iXNlY3AyNTZrMaEDBVt5pk6x0A2fjth25pjLOEE9DpqCG-BCYyvutY04TZ,enr:-HW4QO2vefLueTBEUGly5hkcpL7NWdMKWx7Nuy9f7z6XZInCbFAc0IZj6bsnmj-Wi4ElS6jNa0Mge5Rkc2WGTVemas2AgmlkgnY0iXNlY3AyNTZrMaECR9SmYQ_1HRgJmNxvh_ER2Sxx78HgKKgKaOkCROYwaDY"
+```
+
+### Run with Docker
+
+```shell
+docker run -v /Users/obol/charon/.charon:/opt/charon/.charon obolnetwork/charon:v1.0.0-rc1 alpha test peers \
+  --enrs="enr:-HW4QNDXi9MzdH9Af65g20jDfelAJ0kJhclitkYYgFziYHXhRFF6JyB_CnVnimB7VxKBGBSkHbmy-Tu8BJq8JQkfptiAgmlkgnY0iXNlY3AyNTZrMaEDBVt5pk6x0A2fjth25pjLOEE9DpqCG-BCYyvutY04TZs,enr:-HW4QO2vefLueTBEUGly5hkcpL7NWdMKWx7Nuy9f7z6XZInCbFAc0IZj6bsnmj-Wi4ElS6jNa0Mge5Rkc2WGTVemas2AgmlkgnY0iXNlY3AyNTZrMaECR9SmYQ_1HRgJmNxvh_ER2Sxx78HgKKgKaOkCROYwaDY"
+```
+
+## Beacon
+
+Run tests towards beacon node(s), evaluating the effectiveness of a potential connection of a Charon node running on the same machine.
+
+### Pre-requisites
+
+- Running beacon node(s) towards which tests will be executed.
+
+### Run
+
+```shell
+charon alpha test beacon \
+  --endpoints="https://ethereum-holesky-beacon-api.publicnode.com,https://ethereum-sepolia-beacon-api.publicnode.com"
+```
+
+### Run with Docker
+
+```shell
+docker run obolnetwork/charon:v1.0.0-rc1 alpha test beacon \
```

Comment:
Maybe we shall always provide only docker commands? If I am not mistaken other documentation does just this.

Reply:
I see in some places [like here](https://docs.obol.tech/docs/advanced/quickstart-builder-api#charon) simply charon command is used

Reply:
+1 on always providing docker commands in this case.

Reply:
I think what @pinebit meant was to exclude non-docker commands

Reply:
I don't mind providing both in according with what other articles do.